diff -Naur ./modules/special-mail-tags.php ../contact-form-7/modules/special-mail-tags.php
--- ./modules/special-mail-tags.php	2012-07-31 00:57:05.000000000 -0400
+++ ../contact-form-7/modules/special-mail-tags.php	2012-08-12 20:21:20.000000000 -0400
@@ -67,6 +67,18 @@
 	elseif ( '_post_author_email' == $name )
 		$output = $user->user_email;
 
+	elseif ( 'user_info' == $name ) {
+	  if ( is_user_logged_in() ) {
+	    $output = 'logged in';
+	    $user = wp_get_current_user();
+	        $output = '#' . $user->ID . ' ' . $user->user_login
+		        . " <" . $user->user_firstname . ' ' . $user->user_lastname . ">" 
+		        . " " . $user->display_name
+		  . " " . $user->user_email;
+	  }
+	  else $output = 'User not logged in.';
+	}
+
 	return $output;
 }
 
