diff -Naur ./db-setup_rs.php ../role-scoper/db-setup_rs.php
--- ./db-setup_rs.php	2012-10-10 16:22:10.000000000 -0400
+++ ../role-scoper/db-setup_rs.php	2012-11-13 21:21:14.000000000 -0500
@@ -177,7 +177,10 @@
 	 	KEY role2obj (scope,assign_for,role_type,role_name,src_or_tx_name,obj_or_term_id),
 	 	KEY role2agent (assign_for,scope,role_type,role_name,group_id,user_id),
 	 	KEY role_rs (date_limited,role_type,role_name,scope,assign_for,src_or_tx_name,group_id,user_id,obj_or_term_id),
-	 	KEY role_assignments (role_name,role_type,scope,assign_for,src_or_tx_name,group_id,user_id,obj_or_term_id,inherited_from,assignment_id)
+	 	KEY role_assignments (role_name,role_type,scope,assign_for,src_or_tx_name,group_id,user_id,obj_or_term_id,inherited_from,assignment_id),
+                KEY content_date_limited (content_date_limited),
+                KEY group_id (scope, role_type, group_id),
+                KEY user_id (scope, role_type, user_id)
 	);
 	";
 	
diff -Naur ./terms-interceptor_rs.php ../role-scoper/terms-interceptor_rs.php
--- ./terms-interceptor_rs.php	2012-10-10 16:22:10.000000000 -0400
+++ ../role-scoper/terms-interceptor_rs.php	2012-11-07 18:04:20.000000000 -0500
@@ -155,7 +155,7 @@
 				$return['remap_parents'] = true;
 			}
 
-			if ( is_admin() && ( 's2' == $GLOBALS['plugin_page'] ) ) {
+			if ( is_admin() && ( array_key_exists('plugin_page', $GLOBALS) && 's2' == $GLOBALS['plugin_page'] ) ) {
 				$return['required_operation'] = 'read';
 			}
 		}
